config.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// PushConfig is config for pushing registered metrics to the given URL with the given Interval.
+//
+// URL and Interval are required fields
+type PushConfig struct {
+	// URL defines URL where metrics would be pushed.
+	URL string
+	// Interval determines the frequency of pushing metrics.
+	Interval time.Duration
+
+	// Headers contain optional http request Headers
+	Headers http.Header
+	// ExtraLabels may contain comma-separated list of `label="value"` labels, which will be added
+	// to all the metrics before pushing them to URL.
+	ExtraLabels string
+	// WriteMetricsFn is a callback to write metrics to w in Prometheus text exposition format without timestamps and trailing comments.
+	// See https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#text-based-format
+	WriteMetricsFn func(w io.Writer)
```

Comment:
It is completely unclear the purpose of `WriteMetricsFn` and how to use it properly. I don't think this function must be exposed to users. It looks like this function must be an internal implementation detail.